src/system/group.cr

```diff
@@ -0,0 +1,182 @@
+require "c/grp"
+require "c/sys/limits"
+
+struct System::Group
+  module Limits
+    {% if flag?(:darwin) || flag?(:openbsd) || flag?(:freebsd) %}
+      GID_MAX = LibC::GID_MAX
+    {% elsif flag?(:linux) %}
+      GID_MAX = 0xffffffff_u32
+    {% else %}
+      GID_MAX = 0xffff_u32
+    {% end %}
+    NGROUPS_MAX = LibC::NGROUPS_MAX
+  end
+
+  alias GID = LibC::GidT
+
+  # Returns the group name for the given group ID.
+  # Raises `System::Group::NotFoundError` if no user exists.
+  def self.name(gid : Int) : String
+    group_struct = LibC.getgrgid(int_to_gid(gid))
+    return String.new(group_struct.value.gr_name) if group_struct
```

Comment:
early returns here makes code harder to read, I'd suggest using longer, yet more readable version:

```cr
group_struct = LibC.getgrgid(int_to_gid(gid))
if group_struct
  return ...
end
raise ...
```

Reply:
or just put the raise inside `else`.

Reply:
Tried it, didn't like it. I think its harder to read.